HOK.Elastic.FileSystemCrawler/WorkerEventStream.cs

```diff
@@ -145,77 +149,84 @@ private async Task ActionMove(InputPathEventStream auditEvent)
                     Interlocked.Increment(ref _filesnotfound);
                     return;
                 }
                 ToDoc = DocumentHelper.ReindexTransform(ToDoc);
 
-                if (ildebug) _il.LogDebugInfo("ActionMove OK", auditEvent.Path, auditEvent);
+                if (ildebug) _il.LogDebugInfo("ActionMoveOrCopy OK", auditEvent.Path, auditEvent);
                 if (auditEvent.IsDir)
                 {
                     var fromPublishedPath = PathHelper.GetPublishedPath(auditEvent.PathFrom);
                     #region movedir
                     //first, move any affected Children....actually we should do this regardless if existing doc was found...
-                    var affectedDocuments = _discoveryEndPoint.FindChildren(fromPublishedPath);
+                    var affectedDocuments = _discoveryEndPoint.FindDescendentsForMoving(fromPublishedPath);
                     foreach (var fso in affectedDocuments)
                     {
+                        _ct.ThrowIfCancellationRequested();
                        try
                         {
                             //delete pathfrom doc....
                             string oldPath = fso.Id;
-                            if (fso.IndexName.Equals(FSOdirectory.indexname, StringComparison.OrdinalIgnoreCase) ? !Directory.Exists(oldPath) : !File.Exists(oldPath))
+                            if (auditEvent.PresenceAction == ActionPresence.Move)
                             {
-                                if (Directory.Exists(PathHelper.ContentRoot))//double-check that the filer is online and available and then delete.
+                                if (fso.IndexName.Equals(FSOdirectory.indexname, StringComparison.OrdinalIgnoreCase) ? !Directory.Exists(oldPath) : !File.Exists(oldPath))
                                 {
-                                    Interlocked.Add(ref _deleted, _indexEndPoint.Delete(oldPath, fso.IndexName));
+                                    if (Directory.Exists(PathHelper.ContentRoot))//double-check that the filer is online and available and then delete.
+                                    {
+                                        Interlocked.Add(ref _deleted, _indexEndPoint.Delete(oldPath, fso.IndexName));
+                                    }
                                 }
                             }
                             //add new path doc....
                             string newPublishedPath = fso.Id.Replace(fromPublishedPath, ToDoc.PublishedPath);
                             fso.Id = newPublishedPath;
                             fso.SetFileSystemInfoFromId();
-                            if (ildebug) _il.LogDebugInfo("ActionMove Child", oldPath, newPublishedPath);
-                            fso.Reason = "ActionMove Child";
+                            fso.Acls = SecurityHelper.GetDocACLs(new DirectoryInfo(newPublishedPath));//todo make same change in the movefile region
```

Comment:
Note to self: double-check this and add it if not present @djsiroky

Reply:
Move file will either get the ACLS from the file that has already been moved or throw a Not Found